activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,11 @@
+*   Disable automatic write protection on replicas
+
+    Write protection no longer automatically enable for replicas.
+    Now each application controls when to enable automatic write protection.
```

Comment:
Do you think it's worth mentioning this on https://edgeguides.rubyonrails.org/active_record_multiple_databases.html too?

Reply:
I am reading now, if it was mentioned before we should probably remove it, but I don't know how/if we should mention rails not doing something you maybe didn't know it did at all before.

Reply:
I had a quick read through and couldn't find any mention of this behaviour. I think `AR.while_preventing_writes` could be mentioned but it's not really in the scope of this PR.

Reply:
I can't find a spot that seems like it needs updating, but it is totally possible I am missing it. Let me know if I missed anything @ghiculescu 🙂

Reply:
I would add onto the guides and explain that if you want replicas to act like replicas they need to be configured with readonly users

Reply:
^ that is alluded to, kind of

![image](https://user-images.githubusercontent.com/509837/121720136-a8c91900-caa8-11eb-8ef7-91f6a470cf6a.png)

@HParker I'll make a separate PR, because I don't think I'm making much sense here.

Reply:
Under the setup section we have:

> Second, the username for the writers and replicas should be different, and the replica user's permissions should be set to only read and not write.
> 
> When using a replica database, you need to add a replica: true entry to the replica in the database.yml. This is because Rails otherwise has no way of knowing which one is a replica and which one is the writer.

https://edgeguides.rubyonrails.org/active_record_multiple_databases.html#setting-up-your-application

In the guide already. Should I expand on that?